Tools/CISupport/ews-build/steps.py

```diff
@@ -1456,27 +1456,34 @@ def hideStepIf(self, results, step):
 
 
 class Trigger(trigger.Trigger):
-    def __init__(self, schedulerNames, include_revision=True, triggers=None, **kwargs):
+    def __init__(self, schedulerNames, include_revision=True, triggers=None, patch=False, pull_request=False, **kwargs):
         self.include_revision = include_revision
         self.triggers = triggers
-        set_properties = self.propertiesToPassToTriggers() or {}
+        set_properties = self.propertiesToPassToTriggers(patch=patch, pull_request=pull_request) or {}
         super(Trigger, self).__init__(schedulerNames=schedulerNames, set_properties=set_properties, **kwargs)
 
-    def propertiesToPassToTriggers(self):
-        properties_to_pass = {
-            'patch_id': properties.Property('patch_id'),
-            'bug_id': properties.Property('bug_id'),
-            'configuration': properties.Property('configuration'),
-            'platform': properties.Property('platform'),
-            'fullPlatform': properties.Property('fullPlatform'),
-            'architecture': properties.Property('architecture'),
-            'owner': properties.Property('owner'),
-        }
-        if self.include_revision:
-            properties_to_pass['ews_revision'] = properties.Property('got_revision')
+    def propertiesToPassToTriggers(self, patch=False, pull_request=False):
+        property_names = [
+            'configuration',
+            'platform',
+            'fullPlatform',
+            'architecture',
+        ]
+        if patch:
+            property_names += ['patch_id', 'bug_id', 'owner']
+        if pull_request:
+            property_names += [
+                'github.base.sha', 'github.head.ref', 'github.head.sha',
+                'github.head.repo.full_name', 'github.number', 'github.title',
+                'repository', 'project', 'owners',
+            ]
         if self.triggers:
-            properties_to_pass['triggers'] = self.triggers
+            property_names.append('triggers')
+
+        properties_to_pass = {prop: properties.Property(prop) for prop in property_names}
         properties_to_pass['retry_count'] = properties.Property('retry_count', default=0)
+        if self.include_revision:
+            properties_to_pass['ews_revision'] = properties.Property('github.base.sha' if pull_request else 'got_revision')
```

Comment:
This seems weird to have github.base.sha here. We should assign this only for patch workflow for now. This can create confusion later-on.

I believe we aren't using ews_revision anywhere in PR work-flow. If we need to use it later-on in PR workflow, let's add it in the same PR which will use it. Also we are already passing github.base.sha directly (in this PR), so we don't need another variable ews_revision for the same thing in PR workflow.